pkg/cmd/roachtest/test_runner.go

```diff
@@ -605,17 +605,22 @@ func (r *testRunner) runWorker(
 			qp.Release(alloc)
 		}
 	}()
-
+	// cluster destroy can be done concurrently. We just need to make sure that we exit the roachtest runner once all
```

Comment:
Nit: comment can be made more precise. The `WaitGroup` just ensures that all pending Destroy calls have completed.

However, note that this may be introducing a new behavior; whereas previously an interrupted worker would exit abruptly, now it has to wait for Destroy(s) to finish.

Reply:
This is correct. Till today, we were relying on GC to destroy the clusters in case of errors.